### PR TITLE
redefine IT digitizer ToF window + add customize fcn for activating IT signal shape with RelVals

### PR DESCRIPTION
**Pull request by emiglior**
#### PR description:

This PR concerns the phase-2 upgrade (Inner Tracker). 

Motivation: The PR improves the description of the data rates in output from the Inner Tracker (IT) and it allows the central production of PU200 RelVals for the studies carried on by the phase-2 Tracker DPS group. 

This PR updates the configuration of the PixelDigitizerAlgorithm and of the Pixel3DDigitizerAlgorithm used for the IT:
1. it updates the [tofLowerCut, tofUpperCut] selection from [-12.5,+12.5] ns to a more sensible window [-5,+20] ns;
2. it introduces a process modifier to activate the emulation of the signal shape of the IT readout chip (already in CMSSW, but OFF by default);
3. it introduces the corresponding workflows with suffix xxxxx.141.  The default workflows (xxxxx.0) have still the emulation of IT signal shape OFF.

The effect of the PR has been discussed in the Tracker Phase-2 simulation meeting:
- motivation for the choice of the new tof window: slide 3 of [this talk](https://indico.cern.ch/event/1242571/contributions/5222206/subcontributions/420756/attachments/2622169/4533947/em20230331.pdf)
-  effect on the occupancy of the new tof window or of the activation of the IT signal shape:  slide 1 of [this talk](https://indico.cern.ch/event/1242571/contributions/5222206/subcontributions/420756/attachments/2622169/4538628/em20230331-extras.pdf). On average, the effect on the occupancy (DIGI level) due to this change is expected to be limited.

#### PR validation:

The PR has been validated running the phase-2 w/f 25061.0 (reference) and 25061.141 (IT signal shape ON).

It can be tested with the w/f 24834.0 (default) and 24834.141 (IT signal shape ON)

#### 

This PR is not a backport.

@suchandradutta @mmusich @zdemirag

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * relvals_opt= -w upgrade
   * workflows = 24834.0, 24834.141

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -1848,6 +1848,29 @@ def condition(self, fragment, stepList, key, hasHarvest):
...
+    steps =  [
+        'Digi',
+        'DigiTrigger',
+    ],
+    PU =  [
+        'Digi',
+        'DigiTrigger',
+    ],
+    suffix = '_ITSignalShape',
+    offset = 0.141
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:1871]: would you mind adding for doc purpose this also in the README file of the package:

https://github.com/cms-sw/cmssw/tree/master/Configuration/PyReleaseValidation|

(also adding it for the OT inefficiency workflow - that I forgot when introducing them would be good).

File: SimTracker/SiPhase2Digitizer/python/customizeForPhase2TrackerSignalShape.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#
+# activate signal shape in IT only
+#
+
+def customizeSiPhase2ITSignalShape(process):
+    ## for standard mixing
+    if hasattr(process,'mix') and hasattr(process.mix,'digitizers') and hasattr(process.mix.digitizers,'pixel'): 
+        if hasattr(process.mix.digitizers.pixel,'PixelDigitizerAlgorithm'):
+            print("# Activating signal shape emulation in IT pixel (planar)")
+            process.mix.digitizers.pixel.PixelDigitizerAlgorithm.ApplyTimewalk = cms.bool(True)
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/python/customizeForPhase2TrackerSignalShape.py:12]:
```suggestion
            process.mix.digitizers.pixel.PixelDigitizerAlgorithm.ApplyTimewalk = True
```

**Review by mmusich — COMMENTED** [review]
some *minor* suggestion, considering these kind of changes are generally required:
   * document the new workflows in the `Configuration/PyReleaseValidation` package
   * drop cms type specifications when possible.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
@emiglior @mmusich 
Any idea why we see failure in Run-3 DD4hep XML workflow? If not, should the test be re-trigger, to make sure it clean? Thx.

**Comment by mmusich** [comment]
>Any idea why we see failure in Run-3 DD4hep XML workflow comparison? 

As far as I know that wf has not a very good track record of stability

**Comment by srimanob** [comment]
Thanks @mmusich 
From your statement, it seems strange because the 11634 is the main workflow we use for Run-3. Using XML (.911) or DB (.0) should not give different result (i.e. instability of tracking). I don't see many failure of 11634.911 in PR test (under upgrade). Could you please point to any discussion or something happened before?

I don't want to hold the approval, just to make sure we understand the effect, specially on something we don't expect it to effect. Thanks for understanding.

**Comment by mmusich** [comment]
@srimanob 

> Could you please point to any discussion or something happened before?

See please https://github.com/cms-sw/cmssw/issues/41200

> I don't want to hold the approval, just to make sure we understand the effect, specially on something we don't expect it to effect. 

Up to you to decide, I am not pushing for this PR. Feel free to retrigger tests, these changes cannot possibly cause any change in Run3.

**Comment by makortel** [comment]
@cmsbuild, please test

Let's try again. I don't think we have seen that massive differences in 11634.911 recently (or at least any mentions were not findable with GitHub search; https://github.com/cms-sw/cmssw/issues/41200 is quite localized in terms of DQM histograms, and turned out to occur also in other workflows). It could be good to document this apparent non-reproducibility in a separate issue (we have seen some cases on non-reprocibility in SIM like in https://github.com/cms-sw/cmssw/issues/34448)

**Comment by emiglior** [comment]
> OK, the new test shows that the failure in Run-3 workflow is gone. Only Phase-2 workflows that show difference. I think this is expected.

I just want to confirm that the modifier introduced (.141) is supposed to run only in phase-2 w/f (it is checked against the presence of the phase-2 InnerTracker digitizers)

**Comment by sunilUIET** [comment]
+pdmv

**Comment by perrotta** [comment]
+1

**Pull request merged**